### PR TITLE
Feature/aave comp inst transfer

### PR DESCRIPTION
**Pull request by majkic99**

### CODE REVIEW

File: contracts/actions/compound/CompBorrow.sol [hunk 1]
```diff
@@ -58,17 +57,13 @@ contract CompBorrow is ActionBase, CompHelper {
...
         // if the tokens are borrowed we need to enter the market
         enterMarket(_cTokenAddr);
-
         require(ICToken(_cTokenAddr).borrow(_amount) == NO_ERROR, ERR_COMP_BORROW);
-
         // always return WETH, never native Eth
         if (tokenAddr == TokenUtils.WETH_ADDR) {
             TokenUtils.depositWeth(_amount);
         }
-
```
Comment (yippee-ki-yay) [inline contracts/actions/compound/CompBorrow.sol:71]: Why you have space ser?

File: contracts/actions/compound/CompPayback.sol [hunk 2]
```diff
@@ -56,25 +65,31 @@ contract CompPayback is ActionBase, CompHelper {
...
+        uint256 maxDebt = ICToken(_cTokenAddr).borrowBalanceCurrent(_onBehalf);
         _amount = _amount > maxDebt ? maxDebt : _amount;
 
         tokenAddr.pullTokensIfNeeded(_from, _amount);
 
         // we always expect actions to deal with WETH never Eth
         if (tokenAddr != TokenUtils.WETH_ADDR) {
             tokenAddr.approveToken(_cTokenAddr, _amount);
-            require(ICToken(_cTokenAddr).repayBorrow(_amount) == NO_ERROR, ERR_COMP_PAYBACK_FAILED);
+            require(ICToken(_cTokenAddr).repayBorrowBehalf(_onBehalf ,_amount) == NO_ERROR, ERR_COMP_PAYBACK_FAILED);
```
Comment (yippee-ki-yay) [inline contracts/actions/compound/CompPayback.sol:86]: Are we sure behavior is exactly the same in `repayBorrow` and `repayBorrowBehalf `

File: contracts/actions/insta/InstPullTokens.sol [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    function _pullTokens(Params memory _inputData) internal {
+        require (_inputData.to != address(0), "Receiver address can't be burn address");
+        bytes memory spellData = _createSpell(_inputData);
+        (bool success, ) = _inputData.dsaAddress.call(spellData);
+
+        require(success, "fallback function call failed");
```
Comment (yippee-ki-yay) [inline contracts/actions/insta/InstPullTokens.sol:55]: Maybe a more descriptive err msg like 'withdraw from dsa failed'

File: contracts/actions/compound/CompGetDebt.sol [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import "../ActionBase.sol";
+import "../../utils/TokenUtils.sol";
+import "../../interfaces/compound/ICToken.sol";
+
+/// @title Action that gets debt amount for a single asset on Compound for debtor
+contract CompGetDebt is ActionBase {
+    using TokenUtils for address;
+
+    struct Params {
+        address tokenAddr;
```
Comment (yippee-ki-yay) [inline contracts/actions/compound/CompGetDebt.sol:14]: cTokenAddr

File: test/recipes/inst-shifter/inst-comp-debt-shift.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+            [CCOMP_ADDR, CUNI_ADDR],
+            [true, true],
+        );
+        const compWithdrawAction = new dfs.actions.compound.CompoundWithdrawAction(
+            CETH_ADDR,
+            '$1',
+            dydxFlAddr,
+        );
+        // repay flashloan
+        const transferRecipe = new dfs.Recipe('TransferDebtlessAaveFromInstadapp', [
```
Comment (yippee-ki-yay) [inline test/recipes/inst-shifter/inst-comp-debt-shift.js:125]: TransferDebtlessAaveFromInstadapp is Compound

Comment (yippee-ki-yay) [reply test/recipes/inst-shifter/inst-comp-debt-shift.js:125]: Also it has debt

File: contracts/actions/compound/CompCollateralSwitch.sol [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import "../../utils/TokenUtils.sol";
+import "../ActionBase.sol";
+import "./helpers/CompHelper.sol";
+
+/// @title Switch if you'll use tokens for collateral on compound
+contract CompCollateralSwitch is ActionBase, CompHelper {
+    using TokenUtils for address;
+
+    struct Params {
+        address[] tokens;
```
Comment (yippee-ki-yay) [inline contracts/actions/compound/CompCollateralSwitch.sol:15]: cTokens

File: contracts/actions/insta/InstPullTokens.sol [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+pragma solidity =0.7.6;
+pragma experimental ABIEncoderV2;
+
+import "./../ActionBase.sol";
+import "../../utils/TokenUtils.sol";
+import "../../DS/DSMath.sol";
+import "../../interfaces/mcd/IManager.sol";
+
+// @notice Pull tokens from DSA
```
Comment (yippee-ki-yay) [inline contracts/actions/insta/InstPullTokens.sol:11]: /// @title (title that should describe the)

**Review by yippee-ki-yay — APPROVED** [review]
Lgtm only few small changes needed

File: contracts/actions/compound/CompPayback.sol (hunk 2 above)
Comment (majkic99) [reply contracts/actions/compound/CompPayback.sol:86]: according to their docs yes https://compound.finance/docs/ctokens#repay-borrow-behalf

**Pull request merged**